tests/50federation/51transactions.pl

```diff
@@ -55,31 +55,62 @@
 # sent. It is unclear if this is the correct behavior or not.
 #
 # See https://github.com/matrix-org/synapse/issues/7543
-test "Server rejects invalid JSON in a version 6 room",
+test "Server discards events with invalid JSON in a version 6 room",
    requires => [ $main::OUTBOUND_CLIENT,
                  federated_rooms_fixture( room_opts => { room_version => "6" } ) ],
 
    do => sub {
       my ( $outbound_client, $creator, $user_id, @rooms ) = @_;
 
       my $room = $rooms[0];
+      my $room_id = $room->room_id;
+
+      my $good_event = $room->create_and_insert_event(
+          type => "m.room.message",
+
+          sender  => $user_id,
+          content => {
+              body    => "Good event",
+          },
+      );
 
       my $bad_event = $room->create_and_insert_event(
           type => "m.room.message",
 
           sender  => $user_id,
           content => {
-             body    => "Message 1",
+             body    => "Bad event",
              # Insert a "bad" value into the PDU, in this case a float.
              bad_val => 1.1,
           },
       );
 
-      my @pdus = ( $bad_event );
+      my @pdus = ( $good_event, $bad_event );
 
       # Send the transaction to the client and expect a fail
       $outbound_client->send_transaction(
           pdus => \@pdus,
           destination => $creator->server_name,
-      )->main::expect_m_bad_json;
+      )->then(sub {
+         # Wait for the good event to be sent down through sync
+         await_sync_timeline_contains( $creator, $room_id, check => sub {
+            my ( $event ) = @_;
+            $event->{type} eq "m.room.message" &&
+               $event->{content}{body} eq "Good event"
+         })
```

Comment:
Can we also check that we *don't* receive the bad event down sync?

Reply:
I couldn't find utilities to make sure that the sync timeline *doesn't* contain something :(

Reply:
Wouldn't it be just adding `&& $event->{content}{body} neq "Good event"`?

Reply:
It only checks if one event in the timeline checks that constraint